### PR TITLE
Fix summoning boat logic in relation to AI

### LINKED ISSUE #3489: Adventure map, Summon Boat spell, boat can be summoned at the same location as an AI boat occupied by an opposing hero
In fheroes2, a boat can be summoned to the same location as an AI boat occupied by an opposing hero.

In fheroes2:

https://user-images.githubusercontent.com/43583994/119264961-5a310900-bbe5-11eb-87a8-c99c057daee3.mp4

Here is the save file:
[Before summon boat over ai b.zip](https://github.com/ihhub/fheroes2/files/6528261/Before.summon.boat.over.ai.b.zip)

### PR DESCRIPTION
**Pull request by vasilenkoalexey**
Fix #3489

### CODE REVIEW

File: src/fheroes2/maps/maps.cpp [hunk 1]
```diff
@@ -330,13 +330,19 @@ void Maps::ClearFog( s32 index, int scoute, int color )
     }
 }
 
-Maps::Indexes Maps::ScanAroundObject( s32 center, int obj )
+Maps::Indexes Maps::FindBoatTiles( int32_t center )
```
Comment (ihhub) [inline src/fheroes2/maps/maps.cpp:333]: Let's please not to create a separate function just for one case as we could have more cases in the future. What if we modify the original `ScanAroundObject` function to accept an extra boolean argument `const bool ignoreHeroes` without default value and then modify the existing calls which are only 6 cases. To differentiate 2 functions let's call the one with distance as `ScanAroundObjectWithDistance`.

**Review by ihhub — CHANGES_REQUESTED** [review]
Hi @vasilenkoalexey , I left just one comment with different approach to solve the problem. Could you please check it?

File: src/fheroes2/maps/maps.cpp (hunk 1 above)
Comment (vasilenkoalexey) [reply src/fheroes2/maps/maps.cpp]: Done

**Comment by ihhub** [comment]
Hi @Branikolog , could you please test this issue?

**Comment by Branikolog** [comment]
Hi, @ihhub !
With this fix we cannot summon boat over other one anymore.
But I've found another issue: we can summon opponent's boat, after he disembarked.
image
image
image

[Summon Opponents boat.zip](https://github.com/ihhub/fheroes2/files/6638682/Summon.Opponents.boat.zip)
(Disembark by a green hero. End turn. Summon boat by Vatavna.

**Comment by vasilenkoalexey** [comment]
I think issue should be renamed to "... relation to another hero on boat"

> But I've found another issue: we can summon opponent's boat, after he disembarked.

We shouldn't be able to? Now the boat has no reference to the owner or creator.

**Comment by Branikolog** [comment]
Any boat, that once was summoned or embarked by a hero is considered as property of a player and cannot be summoned, until other player embarks it and refreshes ownership.

**Comment by vasilenkoalexey** [comment]
Hmm... Anyway this is a completely different problem.

**Comment by ihhub** [comment]
@vasilenkoalexey , thank you so much for fixing this issue!

**Pull request merged**